### DISCUSSION TITLE
How to redirect all HTTP requests to HTTPS?

### DISCUSSION THREAD

**Discussion by xsinger** [datenstrom/community#272] [Ask a question] [answered]
Hi, I don't know how to write the `.htccess`? My web server with TLS and valid certificate.

My code like this:

```
RewriteCond %{SERVER_PORT} !^443$
RewriteRule ^.*$ https://%{SERVER_NAME}%{REQUEST_URI} [L,R=301]
```
but yellow is not all links redirect http URLs to https, For example: 

>http://abc.com/123 is not redirect to https://abc.com/123
>it redirect to https://abc.com/yellow.php

I do not know why?thx!

**Comment by markseuffert** [comment, marked as answer]
Hi, try the following `.htaccess` configuration:

```
<IfModule mod_rewrite.c>
RewriteEngine on

RewriteCond %{HTTPS} !=on
RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [L,R=301]

DirectoryIndex index.html yellow.php
RewriteRule ^(cache|content|system)/ error [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^ yellow.php [L]
</IfModule>
```

As an alternative, I would recommend to redirect in your virtual hosts configuration:

```
<VirtualHost *:80>
        ServerName example.se
        Redirect 301 / https://example.se/
</VirtualHost>
```

Both alternatives work on my Apache 2.4 server. Hope this helps.

**Comment by xsinger** [comment]
Thank you very much.
This setting has been successful.
Yellow is a great program.
